tests/envs/test_action_dim_check.py

```diff
@@ -17,3 +33,59 @@ def test_serialize_deserialize(environment_id):
 
     with pytest.raises(ValueError, match="Action dimension mismatch"):
         env.step(0.1)
+
+
+@pytest.mark.parametrize("env", make_envs_by_action_space_type(spec_list, Discrete))
+def test_discrete_actions_out_of_bound(env):
+    # Car racing currently allow `out-of-bound` actions
+    if "CarRacing" in env.spec.id:
+        return
+
+    env.reset()
+
+    action_space = env.action_space
+    upper_bound = action_space.start + action_space.n - 1
+
+    with pytest.raises(Exception):
+        env.step(upper_bound + 1)
+
+
+@pytest.mark.parametrize(
+    ("env", "seed"),
+    [(env, 42) for env in make_envs_by_action_space_type(spec_list, Box)],
+)
+def test_box_actions_out_of_bound(env, seed):
+    """Environments with Box actions spaces perform clipping inside `step`.
+    The expected behaviour is that an action `out-of-bound` has the same effect
+    of an action with value exactly at the upper (or lower) bound.
+    """
+    OOB_VALUE = 100
+
+    env.reset(seed=seed)
+
+    oob_env = envs.make(env.spec.id)
+    oob_env.reset(seed=seed)
+
+    dtype = env.action_space.dtype
+
+    upper_bounds = env.action_space.high
+    lower_bounds = env.action_space.low
+
+    for i, (is_upper_bound, is_lower_bound) in enumerate(
+        zip(env.action_space.bounded_above, env.action_space.bounded_below)
+    ):
+        if is_upper_bound:
+            obs, _, _, _ = env.step(upper_bounds)
+            oob_action = upper_bounds.copy()
+            oob_action[i] += np.cast[dtype](OOB_VALUE)
```

Comment:
could you assert that the OOB value is actually out of bounds for the environment first

Reply:
Good, actually find a bug while adding this check :+1: